File: lib/domain/models/prestation.dart
class Prestation {
  final int id;
  final String name;
  final int? priceCents;
  final int? minutes;
  final int? categoryId;
  final DateTime? archivedAt;

  const Prestation({
    required this.id,
    required this.name,
    this.priceCents,
    this.minutes,
    this.categoryId,
    this.archivedAt,
  });

  bool get isArchived => archivedAt != null;
}

File: src/domain/models/prestation.ts
import { z } from 'zod';

export const Prestation = z.object({
  id: z.string(),
  label: z.string(),
  priceCents: z.number().int().nonnegative().nullable(),
  minutes: z.number().int().nonnegative(),
  categoryId: z.string().nullable(),
  isActive: z.boolean(),
  archivedAt: z.string().nullable(),
  ordering: z.number().int(),
});

export type Prestation = z.infer<typeof Prestation>;
